autogluon/task/text_classification/dataset.py

```diff
@@ -1,34 +1,37 @@
-import multiprocessing
+import multiprocessing as mp
 from typing import AnyStr
+import pandas as pd
 
 from mxnet import gluon
 from mxnet.metric import Accuracy, F1, MCC, PearsonCorrelation, CompositeEvalMetric
 import gluonnlp as nlp
 from gluonnlp.data import GlueCoLA, GlueSST2, GlueSTSB, GlueMRPC
 from gluonnlp.data import GlueQQP, GlueRTE, GlueMNLI, GlueQNLI, GlueWNLI
+# from gluonnlp.data.utils import (Splitter, concat_sequence, line_splitter, whitespace_splitter)
 from ...core import *
 from ...utils.dataset import get_split_samplers, SampledDataset
+from ...utils.tabular.utils.loaders import load_pd
 
 __all__ = ['MRPCTask', 'QQPTask', 'QNLITask', 'RTETask', 'STSBTask', 'CoLATask', 'MNLITask',
            'WNLITask', 'SSTTask', 'AbstractGlueTask', 'get_dataset']
 
 @func()
-def get_dataset(path=None, name=None, train=True, *args, **kwargs):
+def get_dataset(name=None, *args, **kwargs):
     """Load a text classification dataset to train AutoGluon models on.
         
         Parameters
         ----------
-        path : str
-            Path to local directory containing text dataset. This dataset should be in GLUE format.
         name : str
             Name describing which built-in popular text dataset to use (mostly from the GLUE NLP benchmark).
             Options include: 'mrpc', 'qqp', 'qnli', 'rte', 'sts-b', 'cola', 'mnli', 'wnli', 'sst', 'toysst'. 
             Detailed descriptions can be found in the file: `autogluon/task/text_classification/dataset.py`
-        train : bool
-            Whether this data will be used for training models.
     """
+    path = kwargs.get('filepath', None)
     if path is not None:
-        raise NotImplementedError
+        if '.tsv' or '.csv' in path:
+            return CustomTSVClassificationTask(*args, **kwargs)
```

Comment:
why `get_dataset` returns `CustomTSVClassificationTask` which sounds like a task

Reply:
1) conceptually it is not just a dataset it has evaluation metric, statics around dataset such as class_labels; 2) to align with the BERT family model training pipeline, especially the necessary data transformation, the information in the task are needed, not only the dataset; 3) to align with the other benchmark dataset implementation in gluonnlp, such as GLUE etc.

Reply:
A dataset is not a task. Why does a dataset have an evaluation metric variable? This should not be the case.

Reply:
Is "task" needed as opposed to "dataset" because you care about handling pairs of sentences for certain tasks like Q&A?  I think we should just hold off on these in the custom dataset case since it can only handle a single text-field for now.

Couldn't all the other things you mentioned be handled in TextClassification.fit(), where the user would specify: evaluation_metric, class_labels (although these should be ideally detected automatically; you can reuse our tabular code for this in `label_cleaner.py`), etc.  

Also I don't think we should try to mold the user-experience/API with a custom dataset after methods specific to GLUE/BERT/GluonNLP.  Rather, I'd say these specific methods should be molded to the custom dataset UX, which is far more important than any one specific benchmark/GLUE task.

Reply:
Task contains dataset, metric and other information needed to train BERT family models both from concept and implementation point of views. If we don't agree on the BERT training processing pipeline, we could either refactor the code for benchmarking datasets. This could yield duplicate codes but function-wise the same. We will need extra amount of work to address this and to align with gluonnlp to enable future pretrained models available in gluonnlp could be directly adapted to autogluon text classification module.

Reply:
Dataset should be entirely separate from task. A task takes a dataset as input to .fit(). A dataset should not know things like evaluation metric, label, split_ratio, etc. Ideally, the dataset should just be a pandas DataFrame or as close to a DataFrame as possible. We shouldn't have to init CustomTSVClassificationTask with any parameters, it should all be passed in the .fit() call, as done in Tabular.

Reply:
1. The task here is an NLP task which aligns with the GLUE task both conceptually and implementation-wise (aligns with the NLP). 2. The nlp task contains dataset class, dataset class doesn't contain evaluation metric.... 3. CustomTSVClassificationTask is an internal implementation. The dataset is processed as pd dataframe but need to convert it to MXNet dataset in order to make it run with mxnet based training pipeline. In tabular, there is no HPO supported, this is different from the scope of NLP. NLP considers HPO as the most basic component in AutoML.

Reply:
2. When task.Dataset() is called, it returns a Task which you have named 'dataset' in your test. With how the code is structured, there is no such thing as a SimpleDataset without a Task that holds it, so effectively to load a dataset, it must also have a metric. This should not be necessary.

3. HPO is supported in Tabular, so I am uncertain what you mean in point 3. Nothing I mentioned was specific to HPO functionality.

Reply:
Regarding the dataset discussion, please refer to the previous response and see the reasons there. We need to have mxnet dataset to enable the dataset to be used in mxnet training script, e.g., the BERT family training scripts in autogluon and gluonnlp. For the metric, please refer to the previous response and see the reasons there. 

It is good that you now support the HPO in tabular. Could you please point out where is the code to support the hyperparameter optimization for tabular nn training, such as model candidates and learning rate range. Thanks.